articles/parametric-vs-nonparametric/index.md

```diff
@@ -0,0 +1,123 @@
+As we shall dissect later, the coefficients of a linear regression function are examples of model parameters. Another example is in the form of the coefficients in logistic regression. In a neural network, the weights act as the parameters.
+
+
+### Parametric Models
+
+Assumptions about the form of a function can ease the process of learning. Parametric models are characterized by simplification of the function to a known form. A parametric model is a learner that summarizes data through a collection of parameters. These parameters are of a fixed-size. This means that the model already knows the number of parameters it requires, regardless of its data. The parameters are also independent of the number of training instances.
+
+With parametric models, there are two steps involved. The first is choosing the function form. Learning the function coefficients from training data is the second step. Let’s expound on the two.
+
+As an example, let’s have the mapping function in the form of a linear regression line.
+
+$$ b_0 + b_1x_1 + b_2x_2 = 0 $$
+
+From the equation, $b_0$, $b_1$, and $b_2$ are coefficients of the line that controls the intercept and slope. Input variables are represented by $x_1$ and $x_2$. As we mentioned before, assumptions about the form of the function make learning easier. Considering the above function, we would need to estimate the coefficients. This results in a predictive model for a given task. Through the use of the intercept and coefficients, we can predict any value. For such a model, feeding in more data will impact the coefficients' value in the above equation. It will not increase the complexity of the model.
+
+It is common to have parametric models being referred to as linear machine learning models. This is as a result of having the assumed form of the function in a linear form. As can be noted from the equation, the combination of input variables is linear. Aside from linear regression, examples of parametric models include logistic regression and linear SVM.
+
+### Non-Parametric Models
+
+Algorithms that do not make particular assumptions about the kind of mapping function are known as non-parametric algorithms. Since they make no assumptions, they can learn any form of the function from data.
+
+One might think that non-parametric means that there are no parameters. However, this is not true. Rather, it simply means that the parameters are not only adjustable but can also change.
+
+[K-nearest neighbors](https://www.geeksforgeeks.org/k-nearest-neighbours/) is an example of a non-parametric algorithm. Its predictions are founded on the k most similar training patterns for a new instance of data. We should note that there are no assumptions made about the form of the mapping function aside from one. It is assumed that most alike' training patterns have a high likelihood of producing a similar output.
```

Comment:
```suggestion
[K-nearest neighbors](https://www.geeksforgeeks.org/k-nearest-neighbours/) is an example of a non-parametric algorithm. Its predictions are based on the k most similar training patterns for a new instance of data. We should note that there are no assumptions made about the form of the mapping function aside from one. It is assumed that most alike' training patterns have a high likelihood of producing a similar output.
```

Can you explain what you mean by assumptions about the mapping function? How are the parameters selected? Explain this part in detail. 

I was having this conversation with one of my classmates. They had doubts about how the parameters are decided and how is the total number of parameters different for various datasets. 

How does one store non-parametric models since it changes with respect to the training data. It may be beneficial to touch upon these topics.